src/core/mac/mac.cpp

```diff
@@ -828,12 +887,17 @@ otError Mac::PrepareDataRequest(TxFrame &aFrame)
     otError  error = OT_ERROR_NONE;
     Address  src, dst;
     uint16_t fcf;
+#if OPENTHREAD_CONFIG_MAC_HEADER_IE_SUPPORT
+    Neighbor *parent = &Get<Mle::MleRouter>().GetParentCandidate();
+#else
+    Neighbor *parent = NULL;
+#endif
 
     SuccessOrExit(error = Get<DataPollSender>().GetPollDestinationAddress(dst));
     VerifyOrExit(!dst.IsNone(), error = OT_ERROR_ABORT);
 
     fcf = Frame::kFcfFrameMacCmd | Frame::kFcfPanidCompression | Frame::kFcfAckRequest | Frame::kFcfSecurityEnabled;
-    UpdateFrameControlField(/* aIsTimeSync */ false, fcf);
+    UpdateFrameControlField(parent, /* aIsTimeSync */ false, fcf);
```

Comment:
This current code is ok as is, but just adding it for a possible future enhancement:

- We get the `parent` neighbor here and pass it to `UpdateFrameControlField()`
- `UpdateFrameControlField()` seems to use `parent` to determine the FCF version (whether to use `kFcfFrameVersion2006` or `kFcfFrameVersion2015`)
- I think would be simpler to just pass version as a param in `UpdateFrameControlField()` instead of a `Neighbor`.
- We can then have a new API in `DataPollSender` to ask it which version should be used for data poll frame

Reply:
This is once discussed. The pros of using such function is that, things like `IsCslEnabled` would only be checked once(otherwise, it needs to be checked to determine the frame version, and then needs to be checked to determine IE present bit. I don't have a strong preference on it. What do you think?

Reply:
The `parent` passed here seems to me not really take any effect in `UpdateFrameControlField()` for CSL_RECEIVER.

Note that `UpdateFrameControlField()` calls `IsCslEnabled()` which wraps the parent version relative check already. 
And `UpdateFrameControlField()` only uses `aNeighbor` info for `CSL_TRANSMITER` to determine the 154 version according the child's sync status.